hibernate-core/src/main/java/org/hibernate/type/descriptor/sql/internal/JdbcLiteralFormatterCharacterData.java

```diff
@@ -34,12 +36,14 @@ public JdbcLiteralFormatterCharacterData(JavaTypeDescriptor javaTypeDescriptor,
 	public String toJdbcLiteral(Object value, Dialect dialect, SharedSessionContractImplementor session) {
 		final String literalValue = unwrap( value, String.class, session );
 
+		final String inlineLiteral = dialect.inlineLiteral( literalValue );
+
 		if ( isNationalized ) {
 			// is there a standardized form for n-string literals?  This is the SQL Server syntax for sure
-			return String.format( Locale.ROOT, "n'%s'", literalValue );
-		}
-		else {
-			return String.format( Locale.ROOT, "'%s'", literalValue );
+			return String.format( Locale.ROOT, NATIONALIZED_FORMAT, inlineLiteral );
```

Comment:
We can simplify the above without the usage of String.format as well. String.format is pretty slow!